expat/lib/xmlparse.c

```diff
@@ -6322,6 +6480,117 @@ normalizePublicId(XML_Char *publicId)
   *p = XML_T('\0');
 }
 
+static LIMIT *
+limitCreate(const XML_Memory_Handling_Suite *ms)
+{
+    LIMIT *limit = (LIMIT *)ms->malloc_fcn(sizeof(LIMIT));
+    if (!limit)
+      return NULL;
+
+    limit->first_entity = NULL;
+    limit->entityNestingLevel = 0;
+    limit->nestedEntityExpansionSize = 0;
+    limit->totalEntityExpansionSize = 0;
+
+    limit->hugeXML = XML_HUGE_XML_DEFAULT ? XML_TRUE : XML_FALSE;
+    limit->nestingLimit = XML_ENTITY_NESTING_LIMIT;
+    limit->maxExpansionSize = XML_ENTITY_EXPANSION_SIZE;
+    limit->expansionRatio = XML_ENTITY_EXPANSION_RATIO;
+    limit->maxHashTableEntries = XML_MAX_HASH_TABLE_ENTRIES;
+
+    return limit;
+}
+
+static void
+limitDestroy(LIMIT *limit,const XML_Memory_Handling_Suite *ms)
```

Comment:
Is `limitDestroy` called at all needed places, yet? I expected a call from `XML_ParserFree`.

Reply:
It's used in two places. I also need to free the structure in case another allocation in ``parserCreate`` fails. But it's just a simple ``FREE``. I'll get rid of the function.